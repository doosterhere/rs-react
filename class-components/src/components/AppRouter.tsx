import { createBrowserRouter, RouterProvider } from 'react-router-dom';

import { ROUTES } from '../constants';

import { HomePage, Page404 } from '../pages';
import { ListItemDetailed } from './ListItemDetailed/ListItemDetailed';

const router = createBrowserRouter([
  {
    path: ROUTES.HOME,
    element: <HomePage />,
    errorElement: <Page404 />,
    children: [
      {
        path: ROUTES.DETAIL,
        element: <ListItemDetailed />,
      },
    ],
  },
]);

const AppRouter = () => <RouterProvider router={router} />;

export { AppRouter };
