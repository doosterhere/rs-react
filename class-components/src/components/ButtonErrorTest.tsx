import { useState } from 'react';

const ButtonErrorTest = () => {
  const [hasError, setHasError] = useState(false);

  const handleThrowError = () => {
    setHasError(true);
  };

  if (hasError) {
    throw new Error('Throwing the error while rendering');
  }

  return (
    <>
      <button type="button" className="button-error" onClick={handleThrowError}>
        Throw an error
      </button>
    </>
  );
};

export default ButtonErrorTest;
