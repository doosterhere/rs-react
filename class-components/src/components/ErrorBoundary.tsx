import { Component, ReactNode } from 'react';

interface IErrorBoundaryProps {
  children: ReactNode;
}

class ErrorBoundary extends Component<IErrorBoundaryProps> {
  state = {
    hasError: false,
    errorMessage: '',
  };

  static getDerivedStateFromError(error: Error) {
    return { hasError: true, errorMessage: error.message.toString() };
  }

  componentDidCatch(error: Error, errorInfo: React.ErrorInfo) {
    console.error(`Error in error boundary: ${error}`, `Component stack: ${errorInfo.componentStack}`);
  }

  reloadPage() {
    window.location.assign('/');
  }

  render() {
    return this.state.hasError ? (
      <div className="container" style={{ display: 'flex', alignItems: 'center' }}>
        <h1 data-testid="error-boundary">Something went wrong!</h1>
        <button onClick={this.reloadPage} style={{ width: '300px' }}>
          Return to main page
        </button>
      </div>
    ) : (
      this.props.children
    );
  }
}

export { ErrorBoundary };
