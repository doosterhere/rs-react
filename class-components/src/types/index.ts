import { PlanetType } from './planet.type';
import { DefaultResponseType } from './default.response.type';

export type { PlanetType, DefaultResponseType };
