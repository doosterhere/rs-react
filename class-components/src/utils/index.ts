import { renderWithRouter } from './test-utils';

export { renderWithRouter };
