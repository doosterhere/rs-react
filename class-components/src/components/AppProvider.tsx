'use client';

import { ReactNode, useRef } from 'react';
import { Provider } from 'react-redux';

import '../styles/global.css';

import { AppStore, configStore } from '../store/store';
import { ThemeProvider } from '../components/ThemeContext';
import { ErrorBoundary } from './ErrorBoundary';

function AppProvider({ children }: { children: ReactNode }) {
  const storeRef = useRef<AppStore>();

  if (!storeRef.current) {
    storeRef.current = configStore();
  }

  return (
    <Provider store={storeRef.current}>
      <ErrorBoundary>
        <ThemeProvider>{children}</ThemeProvider>
      </ErrorBoundary>
    </Provider>
  );
}

export { AppProvider };
