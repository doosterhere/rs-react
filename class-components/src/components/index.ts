import SearchBar from './SearchBar.tsx';
import List from './List.tsx';
import ListItem from './ListItem.tsx';
import Loader from './Loader.tsx';
import ErrorBoundary from './ErrorBoundary.tsx';
import ButtonErrorTest from './ButtonErrorTest.tsx';
import Fallback from './Fallback.tsx';
import Modal from './Modal.tsx';
import AppRouter from './AppRouter.tsx';
import Layout from './Layout.tsx';

export {
  SearchBar,
  List,
  ListItem,
  Loader,
  ErrorBoundary,
  ButtonErrorTest,
  Fallback,
  Modal,
  AppRouter,
  Layout,
};
