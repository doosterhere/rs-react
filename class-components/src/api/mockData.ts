import { DefaultResponseType, PlanetType } from '../types';

const mockDetailedData: PlanetType = {
  name: 'test-planet',
  diameter: '100',
  climate: 'test-climate',
  gravity: '200',
  terrain: 'test-terrain',
  surface_water: '300',
  population: '400',
  url: 'https://swapi.dev/api/planets/1/',
};

const mockData: DefaultResponseType<PlanetType> = {
  count: 2,
  results: [
    {
      name: 'test-planet-1',
      diameter: '100',
      climate: 'test-climate',
      gravity: '200',
      terrain: 'test-terrain',
      surface_water: '300',
      population: '400',
      url: 'https://swapi.dev/api/planets/1/',
    },
    {
      name: 'test-planet-2',
      diameter: '1100',
      climate: 'test-climate',
      gravity: '1200',
      terrain: 'test-terrain',
      surface_water: '1300',
      population: '1400',
      url: 'https://swapi.dev/api/planets/2/',
    },
  ],
};

export { mockDetailedData, mockData };
