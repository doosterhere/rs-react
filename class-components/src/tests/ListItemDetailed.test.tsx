import { screen } from '@testing-library/react';

import { renderWithProvider } from '../utils';
import { ListItemDetailed } from '../components';
import { PlanetType } from '../types';
import { mockedPlanet } from './testSetup/mockData';

describe('ListItemDetailed', () => {
  it('should renders correct data', async () => {
    const { container } = renderWithProvider(<ListItemDetailed data={mockedPlanet} />);

    expect(container).toMatchSnapshot();

    expect(screen.getByText(/Mocked Planet/i)).toBeInTheDocument();
  });

  it('should render correctly with no data received', async () => {
    const { container } = renderWithProvider(<ListItemDetailed data={{} as PlanetType} />);

    expect(container).toMatchSnapshot();

    expect(screen.queryByText(/Mocked Planet/i)).not.toBeInTheDocument();
  });
});
