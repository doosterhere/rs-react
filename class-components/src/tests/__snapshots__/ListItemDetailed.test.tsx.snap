// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItemDetailed should render correctly with no data received 1`] = `
<div>
  <div
    class="content light"
  >
    <div>
      Name: 
    </div>
    <div>
      Diameter: 
    </div>
    <div>
      Gravity: 
    </div>
    <div>
      Population: 
    </div>
    <div>
      Climate: 
    </div>
    <div>
      Terrain: 
    </div>
    <div>
      Surfase water: 
    </div>
    <a
      href="/?search=&page=1"
      role="link"
    >
      Close
    </a>
  </div>
</div>
`;

exports[`ListItemDetailed should renders correct data 1`] = `
<div>
  <div
    class="content light"
  >
    <div>
      Name: 
      Mocked Planet
    </div>
    <div>
      Diameter: 
      10465
    </div>
    <div>
      Gravity: 
      1 standard
    </div>
    <div>
      Population: 
      200000
    </div>
    <div>
      Climate: 
      arid
    </div>
    <div>
      Terrain: 
      desert
    </div>
    <div>
      Surfase water: 
      1
    </div>
    <a
      href="/?search=&page=1"
      role="link"
    >
      Close
    </a>
  </div>
</div>
`;
